src/GithubFilesBundle/Service/GithubImport.php

```diff
@@ -0,0 +1,166 @@
+                $lsDoc->setIdentifier($lsDocIdentifier);
+                $lsDoc->setUri('local:'.$lsDocIdentifier);
+
+                if(empty($this->getValue($lsDocKeys['creator'], $row))){
+                    $lsDoc->setCreator('Imported from GitHub');
+                }else{
+                    $lsDoc->setCreator($this->getValue($lsDocKeys['creator'], $row));
+                }
+
+                $lsDoc->setTitle($this->getValue($lsDocKeys['title'], $row));
+
+                $lsDoc->setOfficialUri($this->getValue($lsDocKeys['officialSourceURL'], $row));
+                $lsDoc->setPublisher($this->getValue($lsDocKeys['publisher'], $row));
+                $lsDoc->setDescription($this->getValue($lsDocKeys['description'], $row));
+                $lsDoc->setVersion($this->getValue($lsDocKeys['version'], $row));
+                $lsDoc->setSubject($this->getValue($lsDocKeys['subject'], $row));
+                $lsDoc->setLanguage($this->getValue($lsDocKeys['language'], $row));
+                $lsDoc->setNote($this->getValue($lsDocKeys['notes'], $row));
+
+                $em->persist($lsDoc);
+                $em->flush();
+                continue;
+            }
+
+            $lsItem = $this->parseGithubStandard($lsDoc, $lsItemKeys, $row);
+            //create a new association between the item and the doc
+        }
+    }
+
+    public function parseGithubStandard(LsDoc $lsDoc, $lsItemKeys, $data){
+        $lsItem = new LsItem();
+        $em = $this->getEntityManager();
+
+        $lsItem->setLsDoc($lsDoc);
+        $lsItem->setFullStatement($this->getValue($lsItemKeys['fullStatement'], $data));
+
+        $lsItemIdentifier = Uuid::uuid5(Uuid::NAMESPACE_URL, $this->getValue($lsItemKeys['fullStatement'], $data))->toString();
+        $lsItem->setIdentifier($lsItemIdentifier);
+        $lsItem->setUri('local:'.$lsItemIdentifier);
+
+        $lsItem->setHumanCodingScheme($this->getValue($lsItemKeys['humanCodingScheme'], $data));
+        $lsItem->setAbbreviatedStatement($this->getValue($lsItemKeys['abbreviatedStatement'], $data));
+        $lsItem->setListEnumInSource($this->getValue($lsItemKeys['listEnumeration'], $data));
+        $lsItem->setConceptKeywords($this->getValue($lsItemKeys['conceptKeywords'], $data));
+        $lsItem->setConceptKeywordsUri($this->getValue($lsItemKeys['conceptKeywordsUri'], $data));
+        $lsItem->setLanguage($this->getValue($lsItemKeys['language'], $data));
+        $lsItem->setLicenceUri($this->getValue($lsItemKeys['license'], $data));
+        $lsItem->setNotes($this->getValue($lsItemKeys['notes'], $data));
+
+        $em->persist($lsItem);
+
+        $lsAssociation = new LsAssociation();
+        $lsAssociation->setLsDoc($lsDoc);
+        $lsAssociation->setOrigin($lsItem);
+        $lsAssociation->setType(LsAssociation::EXACT_MATCH_OF);
+        $lsAssociation->setDestinationNodeIdentifier($lsItemIdentifier);
+
+        $em->persist($lsAssociation);
+        $em->flush();
```

Comment:
This should probably be removed, and let saveGithubDocument() do the one flush for the entire set of entities.